src/main/java/edu/kit/kastel/sdq/lissa/ratlr/classifier/ReasoningClassifier.java

```diff
@@ -99,11 +104,24 @@ private ReasoningClassifier(
     }
 
     @Override
-    protected final Classifier copyOf() {
+    public final Classifier copyOf() {
         return new ReasoningClassifier(
                 threads, cache, provider, prompt, useOriginalArtifacts, useSystemMessage, contextStore);
     }
 
+    @Override
+    public void setClassificationPrompt(String prompt) {
+        this.prompt = prompt;
+    }
+
+    @Override
+    public Map<String, String> getCacheParameters() {
+        Map<String, String> providerParams = provider.getCacheParameters();
+        Map<String, String> params = new HashMap<>(providerParams);
+        params.put("classifier", REASONING_CLASSIFIER_NAME);
```

Comment:
Why does a cache needs to know the classifier? Model, request, seed, model .. should be enough